clippy_lints/src/methods/mod.rs

```diff
@@ -2474,14 +2473,15 @@ fn lint_map_or_none<'a, 'tcx>(
             let map_or_self_snippet = snippet(cx, map_or_args[0].span, "..");
             let map_or_func_snippet = snippet(cx, map_or_args[2].span, "..");
             let hint = format!("{0}.and_then({1})", map_or_self_snippet, map_or_func_snippet);
-            span_lint_and_then(cx, OPTION_MAP_OR_NONE, expr.span, msg, |db| {
-                db.span_suggestion(
-                    expr.span,
-                    "try using `and_then` instead",
-                    hint,
-                    Applicability::MachineApplicable, // snippet
-                );
-            });
```

Comment:
We could write an internal lint for this :thinking:

Reply:
Yeah, it's reasonable to me, filed as #5095.